### ISSUE TITLE
Ktor 3.0.0 support

### ISSUE THREAD

**Issue by yuvaraj119** [kalinjul/kotlin-multiplatform-oidc#75]
**Error 1**
unable to use oidcBearer in install(Auth) config

```
Unresolved reference. None of the following candidates is applicable because of receiver type mismatch:
public fun Auth. oidcBearer(tokenStore: TokenStore, refreshHandler: TokenRefreshHandler, client: OpenIdConnectClient, onRefreshFailed: suspend (Exception) → Unit = ...): Unit defined in org. publicvalue. multiplatform. oidc. ktor
public fun Auth. oidcBearer(tokenStore: TokenStore, refreshAndSaveTokens: suspend (String) → OauthTokens?, onRefreshFailed: suspend (Exception) → Unit): Unit defined in org. publicvalue. multiplatform. oidc. ktor
```

**Error 2** 
```
java.lang.NoClassDefFoundError: Failed resolution of: Lio/ktor/client/plugins/contentnegotiation/ContentNegotiation;
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient$Companion$DefaultHttpClient$2$1.invoke(DefaultOpenIdConnectClient.kt:65)
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient$Companion$DefaultHttpClient$2$1.invoke(DefaultOpenIdConnectClient.kt:64)
                                                                                                    	at io.ktor.client.HttpClientKt.HttpClient(HttpClient.kt:42)
                                                                                                    	at io.ktor.client.HttpClientJvmKt.HttpClient(HttpClientJvm.kt:23)
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient$Companion$DefaultHttpClient$2.invoke(DefaultOpenIdConnectClient.kt:64)
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient$Companion$DefaultHttpClient$2.invoke(DefaultOpenIdConnectClient.kt:63)
                                                                                                    	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient$Companion.getDefaultHttpClient(DefaultOpenIdConnectClient.kt:63)
                                                                                                    	at org.publicvalue.multiplatform.oidc.DefaultOpenIdConnectClient.<init>(DefaultOpenIdConnectClient.kt:58)
                                                                                                    	at org.publicvalue.multiplatform.oidc.OpenIdConnectClientKt.OpenIdConnectClient(OpenIdConnectClient.kt:32)
```

**Comment by realdadfish** [comment]
FYI I stumbled upon the same issue, however, [looking at the changelog of 3.0.0](https://ktor.io/changelog/3.0/) there is no mention of binary breakage, so I created an upstream ticket to get this eventually fixed: https://youtrack.jetbrains.com/issue/KTOR-7579

**Comment by amirplaza** [comment]
Hi! I made some progress on my branch.
https://github.com/amirplaza/kotlin-multiplatform-oidc/tree/feature/ktor-3%2Ckotlin-2.0.21

**Comment by kalinjul** [comment]
Could you check 0.12.0-SNAPSHOT?
To use a snapshot version, you must add maven("https://s01.oss.sonatype.org/content/repositories/snapshots/") to your repositories. 
See [available snapshots](https://s01.oss.sonatype.org/content/repositories/snapshots/io/github/kalinjul/kotlin/multiplatform/oidc-appsupport/).

**Comment by kalinjul** [comment]
ktor 3 support is available in 0.12.0
